packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -126,7 +126,41 @@ class _ProductImageViewerState extends State<ProductImageViewer> {
                   child: Padding(
                     padding:
                         const EdgeInsets.symmetric(horizontal: SMALL_SPACE),
-                    child: _getUnselectImageButton(appLocalizations),
+                    child: (imageProvider == null)
+                        ? Container()
+                        : EditImageButton(
```

Comment:
I cannot see the added value to ignore method `_getUnselectImageButton` and put some new code here.
You'd be better off putting your new code inside `_getUnselectImageButton` instead.
For code consistency.

PS: and in addition to that the tests fail because of that.